### ISSUE TITLE
Add Support for Deconvolution Layer for Keras model

### ISSUE THREAD

**Issue by dsmilkov** [tensorflow/tfjs#53]
_From @clover113 on April 2, 2018 8:9_

Currently, when converting keras model with deconvolution2d layer, the converter went fine. However, when importing the model in Tensorflow.js, it outputs "Error: Unknown layer: Conv2DTranspose". 
Also, conv2dtranspose is supported in tensorflow.js.

_Copied from original issue: tensorflow/tfjs-converter#67_

**Comment by dsmilkov** [comment]
_From @carlthome on April 2, 2018 9:56_

Similarly when going through the SavedModel converter with [`tf.layers.conv2d_transpose`](https://www.tensorflow.org/api_docs/python/tf/layers/conv2d_transpose) you'll end up with 
```
Unsupported Ops in the model
StridedSlice
```

**Comment by dsmilkov** [comment]
_From @clover113 on April 2, 2018 9:59_

Yeah, it seems that tensorflow.js's support for layers is too limited. But it is not good not to have Conv2dTransposed layer.

**Comment by dsmilkov** [comment]
_From @nsthorat on April 2, 2018 13:37_

@caisq can we add this layer next? The op exists in core in forward mode (we'll eventually get to the backprop for it but it's non-trivial).

**Comment by dsmilkov** [comment]
_From @caisq on April 2, 2018 13:58_

@clover113 Thanks for filing the bug! We can add this layer, but it'll support inference first, without supporting training because the gradients for conv2DTranspose is still missing.

**Comment by clover113** [comment]
Now the deconvolution itself seems to work. But I still cannot import the model. It says: "Error: Could not find weights in manifest with names: deconvolution/kernel_1". I only have one deconvolution layer and I import the model using tf.loadModel() only once. Other layers are just fine, so I think the implementation of importing keras model with deconvolution layer is still broken.

**Comment by caisq** [comment]
@clover113 Can you let us know what Keras model you are trying to import, so we can try to reproduce the error on our end?

**Comment by oeway** [comment]
+1 support training for conv2DTranspose
